acclimatise/model.py

```diff
@@ -370,31 +370,59 @@ def capital(self):
         )
 
 
-int_re = re.compile("(int(eger)?)|size|length|max|min", flags=re.IGNORECASE)
-str_re = re.compile("str(ing)?", flags=re.IGNORECASE)
-float_re = re.compile("float|decimal", flags=re.IGNORECASE)
-bool_re = re.compile("bool(ean)?", flags=re.IGNORECASE)
-file_re = re.compile("file|path", flags=re.IGNORECASE)
-dir_re = re.compile("folder|directory", flags=re.IGNORECASE)
+int_re = re.compile(
+    r"\b((int(eger)?)|size|length|max|min|(num(ber)?))\b", flags=re.IGNORECASE
+)
+str_re = re.compile(r"\bstr(ing)?\b", flags=re.IGNORECASE)
+float_re = re.compile(r"\b(float|decimal)\b", flags=re.IGNORECASE)
+bool_re = re.compile(r"\bbool(ean)?\b", flags=re.IGNORECASE)
+file_re = re.compile(r"\b(file|path)\b", flags=re.IGNORECASE)
+input_re = re.compile(r"input", flags=re.IGNORECASE)
+output_re = re.compile(r"output", flags=re.IGNORECASE)
+dir_re = re.compile(r"\b(folder|directory)\b", flags=re.IGNORECASE)
+
+float_num_re = re.compile(
+    r"[+-]?(([0-9]*\.[0-9]+)|((?:0|[1-9]\d*)(?:\.\d*)?(?:[eE][+\-]?\d+)))",
+    flags=re.IGNORECASE,
+)
+int_num_re = re.compile(r"([+-]?[0-9]+)", flags=re.IGNORECASE)
+
+
+def distinguish_inout(string, cls) -> cli_types.CliFileSystemType:
+    """
+    distinguish input/output files/directories given a string
+    """
+    im = input_re.search(string)
+    om = output_re.search(string)
+    if im and not om:
+        return cls(output=False)
+    elif not im and om:
```

Comment:
Also here I don't think it makes sense to have 3 cases. Either it is an output, or it isn't.